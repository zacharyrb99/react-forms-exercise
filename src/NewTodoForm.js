import {useState} from 'react';

const NewTodoForm = ({add}) => {
    const INITIAL_STATE = {
        text: ''
    };

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        add({...formData});
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor='text'> To-Do Text: </label>
            <input id='text' type='text' name='text' placeholder='To-Do Text' value={formData.text} onChange={handleChange}/>

            <button>Add To-Do</button>
        </form>
    )
}

export default NewTodoForm;