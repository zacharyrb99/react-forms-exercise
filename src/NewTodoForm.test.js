import {render} from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('NewTodoForm renders without crashing', () => {
    render(<NewTodoForm/>);
});

test('NewTodoForm matches snapshot', () => {
    const {asFragment} = render(<NewTodoForm/>);
    expect(asFragment).toMatchSnapshot();
});