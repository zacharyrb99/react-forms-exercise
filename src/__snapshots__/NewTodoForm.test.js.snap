// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm matches snapshot 1`] = `[Function]`;
